packages/dcos-integration-test/extra/test_legacy_user_management.py

```diff
@@ -0,0 +1,184 @@
+    # legacy OIDC ID Token login method through the 'https://dcos.auth0.com/'
+    # provider. This behavior is maintained in Bouncer for backwards
+    # compatibility.
+    r = dcos_api_session.put('/acs/api/v1/users/user@domain.foo', json={})
+    assert r.status_code == 201, r.text
+
+    # Bouncer annotates the created user (this is new compared to dcos-oauth).
+    r = dcos_api_session.get('/acs/api/v1/users/user@domain.foo')
+    assert r.json()['provider_type'] == 'oidc'
+    assert r.json()['provider_id'] == 'https://dcos.auth0.com/'
+    assert r.json()['is_remote'] is True
+
+    # When the uid however does not appear to be an email address the more sane
+    # behavior of Bouncer takes effect: an empty (meaningless) JSON body
+    # results in a useful error message.
+    r = dcos_api_session.put('/acs/api/v1/users/user1', json={})
+    assert r.status_code == 400
+    assert 'One of `password` or `public_key` must be provided' in r.text
+
+
+@pytest.mark.usefixtures('remove_users_added_by_test')
+def test_user_put_email_uid_and_description(dcos_api_session):
+    r = dcos_api_session.put(
+        '/acs/api/v1/users/user1@domain.foo',
+        json={'description': 'integration test user'}
+    )
+    assert r.status_code == 201, r.text
+
+    users = get_users(dcos_api_session)
+    assert len(users) > 1
+    assert 'user1@domain.foo' in users
+
+
+@pytest.mark.usefixtures('remove_users_added_by_test')
+def test_user_put_with_legacy_body(dcos_api_session):
+    # The UI up to DC/OS 1.12 sends the `creator_uid` and the `cluster_url`
+    # properties although they are not used by dcos-oauth. Bouncer supports
+    # these two properties for legacy reasons. Note(JP): As a follow-up task we
+    # should change the UI to not send these properties anymore, and then remove
+    # the properties from Bouncer's UserCreate JSON schema again, ideally within
+    # the 1.13 development cycle.
+    r = dcos_api_session.put(
+        '/acs/api/v1/users/user2@domain.foo',
+        json={'creator_uid': 'any@thing.bla', 'cluster_url': 'foobar'}
+    )
+    assert r.status_code == 201, r.text
+
+
+@pytest.mark.usefixtures('remove_users_added_by_test')
+def test_user_conflict(dcos_api_session):
+    # Note: the empty request body is not the decisive criterion here.
+    r = dcos_api_session.put('/acs/api/v1/users/user2@domain.foo', json={})
+    assert r.status_code == 201, r.text
+
+    r = dcos_api_session.put('/acs/api/v1/users/user2@domain.foo', json={})
+    assert r.status_code == 409, r.text
+
+
+@pytest.mark.usefixtures('remove_users_added_by_test')
+def test_user_delete(dcos_api_session):
+    r = dcos_api_session.put('/acs/api/v1/users/user6@domain.foo', json={})
+    r.raise_for_status()
+    assert r.status_code == 201
+
+    r = dcos_api_session.delete('/acs/api/v1/users/user6@domain.foo')
+    r.raise_for_status()
+    assert r.status_code == 204
+
+    users = get_users(dcos_api_session)
+    assert 'user6@domain.foo' not in users
```

Comment:
Another interesting thing we could test here: does adding `user6@domain.foo` again give a 201 or a 409?

Reply:
You are right, but from a higher level point of view this is tested elsewhere and if we want to test things like that in the DC/OS integration test suite then these tests should go in to e.g. test_iam.py.